src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-groups/osd-devices-selection-groups.component.html

```diff
@@ -0,0 +1,43 @@
+<!-- button -->
+<div class="form-group row">
+  <label class="col-sm-3 col-form-label"
+         for="createDeleteButton">
+    <ng-container i18n>{{ name }} devices</ng-container>
+    <cd-helper>
+      <span i18n
+            *ngIf="type === 'data'">The primary storage devices. These devices contain all OSD data.</span>
+      <span i18n
+            *ngIf="type === 'wal'">Write-Ahead-Log devices. These devices are used for BlueStore’s internal journal. It is only useful to use a WAL device if the device is faster than the primary device (e.g. NVME devices or SSDs). If there is only a small amount of fast storage available (e.g., less than a gigabyte), we recommend using it as a WAL device.</span>
+      <span i18n
+            *ngIf="type === 'db'">DB devices can be used for storing BlueStore’s internal metadata.  It is only helpful to provision a DB device if it is faster than the primary device (e.g. NVME devices or SSD).</span>
+    </cd-helper>
+  </label>
+  <div class="col-sm-9">
+    <ng-container *ngIf="devices.length === 0; else blockClearDevices">
+      <button type="button"
+              class="btn btn-light"
+              (click)="showSelectionModal()"
+              [disabled]="availDevices.length === 0 || !canSelect">
+        <i [ngClass]="[icons.add]"></i>
+        <ng-container i18n>Add</ng-container>
+      </button>
+    </ng-container>
+    <ng-template #blockClearDevices>
+      <div class="pb-2 my-2 border-bottom">
+        <span *ngFor="let filter of appliedFilters">
+          <span class="badge badge-dark mr-2">{{ filter.label }}: {{ filter.formatValue }}</span>
+        </span>
+        <a class="tc_clearSelections"
```

Comment:
![Bildschirmfoto vom 2019-11-22 14-20-21](https://user-images.githubusercontent.com/1897962/69429495-fa2d7e00-0d33-11ea-8cf6-65246c202a87.png)
```suggestion
        <a class="badge badge-danger mr-2 tc_clearSelections"
           href=""
           (click)="clearDevices(); false">
          <ng-container i18n>Clear</ng-container>
          <i [ngClass]="[icons.destroyCircle]"></i>
        </a>
```

Reply:
Thanks! I also tried changing the color of active filters:
`badge-dark`: current design, dark badges are used in many components of dashboard.
![dark](https://user-images.githubusercontent.com/1691518/69517118-bedabb80-0f8e-11ea-957b-9656c9747960.png)

`badge-success`
![green](https://user-images.githubusercontent.com/1691518/69517128-c69a6000-0f8e-11ea-9c4b-5eb6b93bd026.png)

`badge-info`
![info](https://user-images.githubusercontent.com/1691518/69517132-ca2de700-0f8e-11ea-9c7f-00bca43e8c8e.png)

`badge-light`
![light](https://user-images.githubusercontent.com/1691518/69517136-cc904100-0f8e-11ea-9250-661e527bbf86.png)

`badge-warning`
![warning](https://user-images.githubusercontent.com/1691518/69517137-ce5a0480-0f8e-11ea-85e7-c982b81faf2b.png)

Any takes?

Reply:
Everything except the white/bright grey badges. So it's on your choice.

Reply:
Right at the moment i realized another style that we can use instead of the badge clear button. Why not using the GitHub style? Maybe this design is more clear to the user that it can be clicked.

![Bildschirmfoto vom 2019-11-25 08-48-46](https://user-images.githubusercontent.com/1897962/69522011-a2278f00-0f60-11ea-8c30-463c5e146174.png)

Reply:
Here is the icon you can use for that: https://fontawesome.com/icons/window-close?style=solid

Reply:
Thanks, address in https://github.com/ceph/ceph/pull/30921/commits/9f377daddbdd9f98658bb00206cc8c42136fa4e8

![clear](https://user-images.githubusercontent.com/1691518/69532017-ef7f1d00-0faf-11ea-936c-e41885dc26f2.png)

I keep dark badge.

Reply:
:+1: supporting the suggestion by @votdev of using per-filter disables: like [this](https://www.patternfly.org/v3/pattern-library/forms-and-controls/filter/) or [this](https://www.patternfly.org/v4/documentation/core/components/chipgroup).

And dark gray looks good (unless we use some highlight color, like the one for buttons).

Reply:
We can improve tables used to select/view devices. Create an issue to track this https://tracker.ceph.com/issues/43025.